### DISCUSSION TITLE
Issue showing an alert

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3387] [Q&A]
I'm trying to display an alert with the following code:

            .alert(
                alertTitle,
                isPresented: $noName,
                presenting: details
            ) { details in
                Button(role: .cancel) {
                    // ...
                } label: {
                    Text("Delete")
                }
              Button() {
                  // ...
              } label: {
                  Text("retry")
              }
            } message: { details in
                Text(details.error)
            }
            .alert(
                alertTitle,
                isPresented: $noName,
                presenting: details
            ) { details in
                Button(role: .cancel) {
                    // ...
                } label: {
                    Text("cancel")
                }
              Button() {
                  // ...
              } label: {
                  Text("retry")
              }
            } message: { details in
                Text(details.error)
            }

where details just contains an error message.

An alert appears with the alert title and an OK button, but the cance, retry and error message are missing. What am I doing wrong?

(I am using Xcode 16 RC and composable architecture 1.15)

**Comment by mbrandonw** [comment]
Hi @easiwriter, it seems you are using `alert(isPresented:)` twice with the exact same state, which is not correct to do in SwiftUI. That tells SwiftUI to show two alerts when the state flips to `true`.

I am not entirely sure what you are trying to accomplish with this code. It would be best if you provide a full, compiling code snippet that demonstrates the behavior you are seeing. Also, this code does not involve any TCA, and so this seems like a vanilla SwiftUI issue you are having. You may find better help on a more general SwiftUI help forum.

**Reply by easiwriter** [reply]
Thanks for this. I found the answer in the Stacks tour video, so all is OK now.
